onnx/defs/tensor/defs.cc

```diff
@@ -240,7 +240,36 @@ will be (2, 1, 3).
     .TypeConstraint(
         "T",
         {"tensor(float16)", "tensor(float)", "tensor(double)"},
-        "Constrain input and output types to float tensors.");
+        "Constrain input and output types to float tensors.")
+    .ShapeInferenceFunction([](InferenceContext& ctx) {
+    std::vector<int64_t> perm;
+    {
+      auto perm_attr = ctx.getAttribute("perm");
```

Comment:
I wonder whether we can have a simple wrapper that typecasts the attribute values to the right type. It'd make implementing functions much easier. Those helpers might be more isolated that ir.h and be just `getAttr<vector<int>>("perm", optional_default)` or something

Reply:
@dzhulgakov Rip off `GetSingleArgument`/`GetRepeatedArgument` from c2? :-)

Reply:
sure, this is just the initial example though. I'll start adding reusable helpers in the next diff when I start to add more supported operators.